packages/pigeon/lib/java_generator.dart

```diff
@@ -171,7 +171,13 @@ static MessageCodec<Object> getCodec() {
                   indent.writeln(
                       'ArrayList<Object> args = (ArrayList<Object>)message;');
                   enumerate(method.arguments, (int index, NamedType arg) {
-                    final String argType = _javaTypeForDartType(arg.type);
+                    // The StandardMessageCodec can give us [Integer, Long] for
+                    // a Dart 'int'.  To keep things simple we just use 64bit
+                    // longs in Pigeon with Java.
```

Comment:
Why can't this code use `_castObject`, which already handles this issue?

Reply:
I tried but they are used in slightly different cases.  `_castObject` does `fromMap` which we don't want here.  I tried to split out a shared `_castInt` function but because of oddities of the Java language I couldn't without changing the structure of the generated code.  A cast is fine in that assignment where `_castObject` is being used.  Here it is used in a right hand expression for a function argument which doesn't seem to be allowed.

I also added a platform unit test that mimics this behavior to make sure the other code is working.  I also updated the docstring for `_castObject`.